pkg/controllers/user/multiclusterapp/controller_multiclusterapp.go

```diff
@@ -106,16 +105,16 @@ func (m *MCAppController) sync(key string, mcapp *v3.MultiClusterApp) (runtime.O
 		}
 
 		// check if this app already exists
-		a, err := m.apps.GetNamespaced(projectNS, appNameInProject, metav1.GetOptions{})
-		if err != nil && !apierrors.IsNotFound(err) {
-			return mcapp, fmt.Errorf("error %v for checking if app %v already exists in project %v", err, appNameInProject, projectNS)
-		} else if a != nil && a.Name == appNameInProject {
+		a, err := m.appLister.List(projectNS, set.AsSelector())
```

Comment:
Changing this since we'll no longer know the name of app that will get created. Also, the label of multiclusterappId is set on all apps, so we can list apps in a particular project using that label.